storage/innobase/page/page0page.cc

```diff
@@ -2832,7 +2832,7 @@ page_warn_strict_checksum(
 	ulint				space_id,
 	ulint				page_no)
 {
-	srv_checksum_algorithm_t	curr_algo_nonstrict;
+	srv_checksum_algorithm_t	curr_algo_nonstrict = srv_checksum_algorithm_t();
```

Comment:
OK, but I wonder why any warning is produced in the first place - the switch below is exhaustive. Maybe ut_error is not annotated as "does not return"?

(but please don't spend non-trivial amount of time investigating)

Reply:
The warning is later when the variable is used, and only appears in release mode. The ut_error macro expands to nothing in release builds, does nothing, and then the function used an uninitialized variable.

While it isn't necessarily better now, we at least have a consistent behavior, without using some memory garbage. I wasn't sure how we could implement a release-mode error here, as it's just an assertion failure even in debug builds.